modules/edge-ren/src/lib.rs

```diff
@@ -0,0 +1,464 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use codec::{Encode, Decode, HasCompact};
+use frame_support::{Parameter, decl_error, decl_event, decl_module, decl_storage, ensure, traits::{Get, EnsureOrigin, FungibleAsset, MintableAsset, BurnableAsset}};
+use frame_system::{ensure_none, ensure_signed};
+use edgeware_primitives::Balance;
+use sp_core::ecdsa;
+use sp_io::{crypto::secp256k1_ecdsa_recover, hashing::keccak_256};
+use sp_runtime::{
+	Permill,
+	ModuleId,
+	traits::{Member, StaticLookup, AccountIdConversion},
+	transaction_validity::{
+		InvalidTransaction, TransactionPriority, TransactionSource, TransactionValidity, ValidTransaction,
+	},
+	DispatchResult,
+};
+use sp_std::vec::Vec;
+use coinaddress as btc_address;
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+type EcdsaSignature = ecdsa::Signature;
+type DestAddress = Vec<u8>;
+
+
+
+type TokenIdOf<T> = <<T as Config>::Assets as FungibleAsset<<T as frame_system::Config>::AccountId>>::AssetId;
+type BalanceOf<T> = <<T as Config>::Assets as FungibleAsset<<T as frame_system::Config>::AccountId>>::Balance;
+
+const NAME_MAX_LENGTH : u8 = 32;
+
+pub trait Config: frame_system::Config {
+	type Event: From<Event<Self>> + Into<<Self as frame_system::Config>::Event>;
+	type RenVMTokenIdType: Member + Parameter + Default + Copy + HasCompact;
+	type RenvmBridgeUnsignedPriority: Get<TransactionPriority>;
+	type ControllerOrigin: EnsureOrigin<Self::Origin>;
+	type ModuleId: Get<ModuleId>;
+	type Assets: FungibleAsset<Self::AccountId> + MintableAsset<Self::AccountId> + BurnableAsset<Self::AccountId>;
+}
+
+/// struct				RenTokenInfo
+/// ren_token_id		What this pallet uses to identify tokens ( unique for a given token on this pallet for a given chain )
+/// ren_token_asset_id	How the pallet used for Assets identifies the token
+/// ren_token_name 		Name of the token; used to determine the validation process if any
+/// ren_token_renvm_id	What RenVM uses to uniquely identify this token across different chains
+/// ren_token_pub_key 	The PublicKey used to check the RenVM signature against
+// /// ren_token_proof 	Proof of this token being registered on the RenVM legitimizing it
+/// ren_token_mint_enabled,ren_token_burn_enabled To enable/disable mint/burn temporarily
```

Comment:
This seems unnecessary.

Reply:
Do you mean the "ren_token_proof"? If yes, then I'll get rid of the comments mentioning it. I had put them there a potential field that we might like.